commit 6475ea889e7ae22942ce424a36963ac90b02cef1 (refs/changes/11/3511/2)
Author: Bryan Cantrill <bryan@joyent.com>
Date:   2018-02-28T01:04:33+00:00 (1 year, 7 months ago)
    
    OS-6705 thread spinning in dp_pcache_poll() induces system hang
    Reviewed by: Dan McDonald <danmcd@joyent.com>
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/uts/common/io/devpoll.c b/usr/src/uts/common/io/devpoll.c
index 040a1f9190..e0d37e171b 100644
--- a/usr/src/uts/common/io/devpoll.c
+++ b/usr/src/uts/common/io/devpoll.c
@@ -25,7 +25,7 @@
 
 /*
  * Copyright (c) 2012 by Delphix. All rights reserved.
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 #include <sys/types.h>
@@ -405,6 +405,7 @@ repoll:
 					 * about the fd reassignment.
 					 */
 					pdp->pd_fp = fp;
+					pdp->pd_gen = gen;
 				}
 			}
 			/*
